universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -197,7 +220,35 @@ private ContainerTag renderQuestionDefinitionBox(QuestionDefinition questionDefi
                         .withClasses(Styles.MT_1, Styles.TEXT_SM)));
   }
 
-  private static ContainerTag createScalarDropdown(ImmutableMap<Scalar, ScalarType> scalars) {
+  private ContainerTag createActionDropdown(String blockName) {
+    ImmutableMap<String, String> actionOptions =
+        Arrays.stream(PredicateAction.values())
+            .collect(toImmutableMap(PredicateAction::toDisplayString, PredicateAction::name));
+    return new SelectWithLabel()
+        .setFieldName("predicateAction")
+        .setLabelText(String.format("%s should be", blockName))
+        .setOptions(actionOptions)
+        .getContainer();
+  }
+
+  private Tag createHiddenQuestionDefinitionInput(QuestionDefinition questionDefinition) {
+    return input()
+        .isHidden()
+        .withName("questionId")
+        .withType("number")
+        .withValue(String.valueOf(questionDefinition.getId()));
+  }
+
+  private ContainerTag createScalarDropdown(QuestionDefinition questionDefinition) {
+    ImmutableMap<Scalar, ScalarType> scalars;
+    try {
+      scalars = Scalar.getScalars(questionDefinition.getQuestionType());
+    } catch (InvalidQuestionTypeException | UnsupportedQuestionTypeException e) {
+      // This should never happen since we filter out Enumerator questions before this point.
+      return div()
```

Comment:
question (blocking): This used to return a div instead of the form entirely if we hit this case. Is it ok to change the behavior so the div just appears in place of the scalar dropdown instead? 

Theoretically this should never happen, but if it did it might look a little funky

Reply:
I actually think this is preferable to only replacing the scalar dropdown because we don't want there to be a form at all in this case (don't want the submit functionality, and no need to have the other form fields there in that case). Let me know if you feel strongly another way! I feel comfortable with this especially because like you said, this should never happen.